### PR TITLE
Update Haskell backend submodule version

### PR DESCRIPTION
**Pull request by ehildenb**
Hopefully this will catch breakages in the backend caused by the frontend before they get merged.

### CODE REVIEW

File: k-distribution/tests/regression-new/Makefile [hunk 1]
```diff
@@ -32,7 +32,7 @@ SUBDIRS=issue-2273 \
 	kore-brackets \
 	quadratic-poly-unparsing \
 	io-java \
-	symbolic-int-function \
+	symbolic-int-function
```
Comment (dwightguth) [inline k-distribution/tests/regression-new/Makefile]: this was done deliberately in order to simplify diffs that modify this file, which frequently conflict with one another. I'd appreciate it if it was kept the way it is now.

File: src/main/scripts/test-haskell-backend [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+set -ueo pipefail
+
+rm -rf haskell-backend-kore
+git clone 'https://github.com/kframework/kore' haskell-backend-kore
```
Comment (dwightguth) [inline src/main/scripts/test-haskell-backend]: this doesn't seem at all the right way to do this. you should simply use the source directory of the haskell backend that has already been checked out.

**Review by dwightguth — CHANGES_REQUESTED** [review]

File: k-distribution/tests/regression-new/Makefile (hunk 1 above)
Comment (ehildenb) [reply k-distribution/tests/regression-new/Makefile]: Makes sense, fixed.

**Comment by ehildenb** [comment]
@dwightguth this PR now just updates the Dockerfile and haskell backend submodule (which has new directory structure).

**Review by dwightguth — APPROVED** [review]

**Pull request merged**